### PR TITLE
Extend erdgo

### PR DESCRIPTION
**Pull request by miiu96**
Extend proxy object with new methods:
- GetNetworkEconomics()
- RequestTransactionCost(tx)
- ExecuteVMQuery(request)
- GetTransactionInfoWithResults(txHash)

### CODE REVIEW

File: erdgo/blockchain/interface.go [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+import "github.com/ElrondNetwork/elrond-sdk/erdgo/data"
+
 type addressHandler interface {
 	AddressAsBech32String() string
 	AddressBytes() []byte
 	IsValid() bool
 	IsInterfaceNil() bool
 }
+
+type ProxyHandler interface {
```
Comment (bogdan-rosianu) [inline erdgo/blockchain/interface.go:12]: add comment

File: erdgo/blockchain/proxy.go [hunk 2]
```diff
@@ -147,6 +208,28 @@ func (ep *elrondProxy) GetTransactionInfo(hash string) (*data.TransactionInfo, e
 	return response, nil
 }
 
+func (ep *elrondProxy) RequestTransactionCost(tx *data.Transaction) (*data.TxCostResponseData, error) {
```
Comment (bogdan-rosianu) [inline erdgo/blockchain/proxy.go:211]: add comment

File: erdgo/blockchain/proxy_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-sdk/erdgo/data"
+)
+
+func TestElrondProxy_GetNetworkEconomics(t *testing.T) {
+	t.Skip()
+
+	ep := NewElrondProxy("https://gateway.elrond.com")
```
Comment (bogdan-rosianu) [inline erdgo/blockchain/proxy_test.go:15]: `http://localhost:8079` ?

File: erdgo/blockchain/proxy.go [hunk 4]
```diff
@@ -39,6 +43,30 @@ func NewElrondProxy(url string) *elrondProxy {
 	return ep
 }
 
+// ExecuteVMQuery retrieves data from existing SC trie through the use of a VM
```
Comment (bogdan-rosianu) [inline erdgo/blockchain/proxy.go:46]: As discussed, can you also add a blocking function `WaitForNonce(address string, nonce uint64) error` and perform HTTP requests for the address nonce with a 1-second interval? The function would return `nil` if the received nonce is the same as the one from the parameter or error if timeout or other errors. The timeout could be 2 minutes

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: erdgo/blockchain/proxy.go (hunk 4 above)
Comment (bogdan-rosianu) [reply erdgo/blockchain/proxy.go:46]: + `SimulateTransaction()`

File: erdgo/blockchain/proxy_test.go (hunk 3 above)
Comment (iulianpascalau) [reply erdgo/blockchain/proxy_test.go:15]: this test is skipped, can be left as it is and used only when necessary.

**Review by iulianpascalau — COMMENTED** [review]
Just what Bogdan found

File: erdgo/blockchain/interface.go (hunk 1 above)
Comment (miiu96) [reply erdgo/blockchain/interface.go:12]: done.

File: erdgo/blockchain/proxy.go (hunk 2 above)
Comment (miiu96) [reply erdgo/blockchain/proxy.go:211]: done.

File: erdgo/blockchain/proxy_test.go (hunk 3 above)
Comment (miiu96) [reply erdgo/blockchain/proxy_test.go:15]: done. also test was moved from this directory

File: erdgo/blockchain/proxy.go (hunk 4 above)
Comment (miiu96) [reply erdgo/blockchain/proxy.go:46]: I will add **SimulateTransaction** in another PR when this route will work also this async smart contract calls

File: erdgo/tests/proxy_test.go [hunk 5]
```diff
@@ -42,3 +50,101 @@ func TestGetAccount(t *testing.T) {
...
...nsy","sender":"erd1e6c9vcga5lyhwdu9nr9lya4ujz2r5w2egsfjp0lslrgv5dsccpdsmre6va","data":"ESDTNFTTransfer@595959453643392d303934393030@01@01@080112020001226f08011204746573741a20ceb056611da7c977378598cbf276bc90943a3959441320bff0f8d0ca3618c05b20e8072a206161616161616161616161616161616161616161616161616161616161616161320461626261321268747470733a2f2f656c726f6e642e636f6d3a0474657374@66756e64@57c506a19d8eae1829403a8f3f54e1f0379f38157fcf53d5bca6f237e4a4cf61@1c20","prevTxHash":"c7fadeaccce0673bd6ce3a1f472f7cc1beef20c0b3131cfa9866cd5075816639","originalTxHash":"c7fadeaccce0673bd6ce3a1f472f7cc1beef20c0b3131cfa9866cd5075816639","gasLimit":18250000,"gasPrice":1000000001,"callType":0},{"hash":"f337d2705d2b644f9d8f75cf270e879b6ada51c4c54009e94305adf368d1adbc","nonce":1,"value":102793170000000,"receiver":"erd1e6c9vcga5lyhwdu9nr9lya4ujz2r5w2egsfjp0lslrgv5dsccpdsmre6va","sender":"erd1qqqqqqqqqqqqqpgqtjp7p3pwmn3efaqtynff62vtqfyug8cz396qs5vnsy","data":"@6f6b","prevTxHash":"5ab14959aaeb3a20d95ec6bbefc03f251732d9368711c55c63d3811e70903f4e","originalTxHash":"c7fadeaccce0673bd6ce3a1f472f7cc1beef20c0b3131cfa9866cd5075816639","gasLimit":0,"gasPrice":1000000001,"callType":0}]}},"error":"","code":"successful"}`)
```
Comment (iulianpascalau) [inline erdgo/tests/proxy_test.go:114]: We could have used a struct initialized with fields and then marshaled with the json marshaler. Would have helped the debugging process.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**